File: backend/routers/contacts.py
from typing import List
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session as DBSession
from backend.database.database import get_db
from backend.schemas.contact import ContactResponse, AddContactRequest
from backend.services.contact_service import ContactService
from backend.routers.deps import get_current_user
from backend.models import User

router = APIRouter(prefix="/contacts", tags=["Contacts"])

@router.get("", response_model=List[ContactResponse])
def get_contacts(current_user: User = Depends(get_current_user), db: DBSession = Depends(get_db)):
    return ContactService.get_contacts(db, user_id=current_user.id)

@router.post("", response_model=ContactResponse, status_code=status.HTTP_201_CREATED)
def add_contact(req: AddContactRequest, current_user: User = Depends(get_current_user), db: DBSession = Depends(get_db)):
    return ContactService.add_contact(db, user_id=current_user.id, contact_user_id=req.contact_user_id)

@router.delete("/{contact_user_id}")
def remove_contact(contact_user_id: str, current_user: User = Depends(get_current_user), db: DBSession = Depends(get_db)):
    ContactService.remove_contact(db, user_id=current_user.id, contact_user_id=contact_user_id)
    return {"message": "Contact removed"}

File: frontend/services/api/contacts.ts
import { apiRequest } from './client';
import { User } from '../../types/user';

export interface ContactResponse {
  user_id: string;
  contact_user_id: string;
  contact_user: User;
  created_at: string;
}

export const contactsApi = {
  getContacts: async (): Promise<ContactResponse[]> => {
    return apiRequest<ContactResponse[]>('/contacts');
  },

  addContact: async (contactUserId: string): Promise<ContactResponse> => {
    return apiRequest<ContactResponse>('/contacts', {
      method: 'POST',
      body: JSON.stringify({ contact_user_id: contactUserId }),
    });
  },

  removeContact: async (contactUserId: string): Promise<void> => {
    await apiRequest(`/contacts/${contactUserId}`, {
      method: 'DELETE',
    });
  },
};
